### PR TITLE
MultilineArrayTrailingCommaFixer - fix case with short array on return

### LINKED ISSUE #983: 'multiline_array_trailing_comma' fixer doesn't work with the short syntax
Hi !
The 'multiline_array_trailing_comma' fixer doesn't work if you have :
``` php
return [
	'é' => 5,
	5 => 'é'
];
```

### PR DESCRIPTION
**Pull request by keradus**
> MARK FOR MERGE INTO 2.0 by @keradus
> move logic into transformer

fix #983

### CODE REVIEW

**Comment by keradus** [comment]
PR has been merged.

File: Symfony/CS/Tokenizer/Tokens.php [hunk 1]
```diff
@@ -879,7 +879,7 @@ public function isShortArray($index)
 
         $prevToken = $this[$this->getPrevMeaningfulToken($index)];
 
-        if ($prevToken->equalsAny(array(array(T_DOUBLE_ARROW), '=', '+', '(', '['))) {
+        if ($prevToken->equalsAny(array(array(T_RETURN), array(T_DOUBLE_ARROW), '=', '+', '(', '['))) {
```
Comment (dericofilho) [inline Symfony/CS/Tokenizer/Tokens.php]: This is incomplete - probably as a result of testing for short arrays the wrong way.

It still fails for:
```php
<?php
function a()
{
    yield [
        'é' => 5,
        5 => 'é'
    ];
}
```

Comment (keradus) [inline Symfony/CS/Tokenizer/Tokens.php]: It fixed one case, you found another.
I agree that this solution how we distinguish short vs long array is naive.
Do you know better approach ?

Comment (dericofilho) [inline Symfony/CS/Tokenizer/Tokens.php]: Yes.

```php
if (!$prevToken->equalsAny(array(']', array(T_STRING), array(T_VARIABLE), array(T_ARRAY_CAST))) ) {
    return true;
}
```

**Pull request closed without merging**